src/Todo/App.purs

```diff
@@ -53,127 +53,188 @@ localStorageKey = "todomvc-purescript-state"
 saveState :: State -> Effect Unit
 saveState state = LocalStorage.setItem localStorageKey state
 
-app :: React.Component Props
-app = React.component
-  { displayName: "App"
-  , initialState
-  -- Here we inject a modified `setState` that persists the state to LocalStorage
-  -- after every modification
-  , receiveProps: \a -> receiveProps $ a { setState = \f -> a.setStateThen f saveState }
-  , render:       \a -> render $ a { setState = \f -> a.setStateThen f saveState }
-  }
+data Action
+  = EditNewTodo (Maybe String)
+  | KbdNewTodo
+  | TaskCheck Int
+  | TaskUpdate Int String
+  | TaskDelete Int
+  | ClearCompleted
+  | CheckAllTasks (Maybe Boolean)
+  | UpdateVisibility Visibility
+  | LoadState State
+  | Noop
+
+component :: React.Component Props
+component = React.createComponent "App"
+
+app :: JSX
+app = React.make component
+  { initialState
+  , didMount
+  , render
+  , update
+  , didUpdate
+  } {}
   where
     -- This is the only place we can run stuff only at the first mount
-    receiveProps { state, setState, isFirstMount } = when isFirstMount do
+    didMount self@{ state } = do
+      let setVisibility visibility =
+            React.send self (UpdateVisibility visibility)
       -- On first mount, we start the navigation:
       -- we have something super simple here, in which we match on
       -- the hash string and execute a side effect.
       -- For something fancier we might want to use a parser.
       let matchRoutes hash = case hash of
-            "#/"          -> setState _ { visibility = All }
-            "#/active"    -> setState _ { visibility = Active }
-            "#/completed" -> setState _ { visibility = Completed }
+            "#/"          -> setVisibility All
+            "#/active"    -> setVisibility Active
+            "#/completed" -> setVisibility Completed
             otherwise     -> pure unit
       runEffectFn1 startNavigation matchRoutes
 
       -- Then we try to read if we had some state persisted in LocalStorage
       -- If yes, we overwrite the state with it
       persisted <- LocalStorage.getItem localStorageKey
-      setState \_ -> case persisted of
-        Just (oldState :: State) -> oldState
-        _                        -> state
+      case persisted of
+        Just (oldState :: State) -> React.send self (LoadState oldState)
+        _ -> pure unit
+
+    didUpdate self _ = do
+      saveState self.state
```

Comment:
This is the part that saves to `localStorage` after every update right? It's quite neat :)
Let's move here the comment about saving to `localStorage` after every modification of the state, so it's clearer that's happening here

Reply:
Exactly, this is what updates it =) It was easy to implement too. 

I'll move the comment.